### ISSUE TITLE
ModuleNotFoundError when adding env variable PYTHONIOENCODING on windows

### ISSUE THREAD

**Issue by LeonardoRick** [extrabacon/python-shell#238]
When changing python variables on windows via `env` options object key, a local project installed via `setup.py` is not being recognized  and throw a `ModuleNotFoundError`

**Python code**
```python
import time
import sys
from telegramcustombot.main import Main

class Unbuffered(object):
    def __init__(self, stream):
        self.stream = stream

    def write(self, data):
        self.stream.write(data)
        self.stream.flush()

    def writelines(self, datas):
        self.stream.writelines(datas)
        self.stream.flush()

    def __getattr__(self, attr):
        return getattr(self.stream, attr)


sys.stdout = Unbuffered(sys.stdout)
x = 1
Main('uid', 'botd', 'env', 'email', 'password', 'number', 'altPass')

while True:
    print(f'{x}: test')
    time.sleep(5)
    x += 5
```

**Javascript code**
```javascript
const options = { scriptPath: 'C:\\users\\leona\\iq\\iq-websocket\\python\\botR', env: {  PYTHONIOENCODING : 'utf8'}};

const pyTeste = new PythonShell('teste.py', options);

pyTeste
  .on('message', (line: string) => {
    console.log(line);
  })
  .on('stderr', (std: string) => {
    console.log(std);
  })
  .on('error', err => {
    console.log('err');
    console.log(err);
  })
  .on('close', () => {
    console.log('Programa finalizado');
  });
```

**Expected behavior**
The program should run normaly and recognize the installed module, as it does when running the script alone.

**Actual behavior**
```
PythonShellError: ModuleNotFoundError: No module named 'telegramcustombot'
    at PythonShell.parseError (C:\Users\leona\iq\iq-websocket\node_modules\python-shell\index.ts:333:21)
    at terminateIfNeeded (C:\Users\leona\iq\iq-websocket\node_modules\python-shell\index.ts:191:32)
    at ChildProcess.<anonymous> (C:\Users\leona\iq\iq-websocket\node_modules\python-shell\index.ts:182:13)
    at ChildProcess.emit (events.js:223:5)
    at ChildProcess.EventEmitter.emit (domain.js:475:20)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
    ----- Python Traceback -----
    File "C:\users\leona\iq\iq-websocket\python\botR\teste.py", line 3, in <module>
      from telegramcustombot.main import Main {
  traceback: 'Traceback (most recent call last):\r\n' +
    '  File "C:\\users\\leona\\iq\\iq-websocket\\python\\botR\\teste.py", line 3, in <module>\r\n' +
    '    from telegramcustombot.main import Main\r\n' +
    "ModuleNotFoundError: No module named 'telegramcustombot'\r\n",
  executable: 'py',
  options: null,
  script: 'C:\\users\\leona\\iq\\iq-websocket\\python\\botR\\teste.py',
  args: null,
  exitCode: 1
}
```
**Other Information (please complete the following information):**
 - OS: Windows10
 - Python Version 3.8.7
 - Node Version 12.15.0

**Comment by Almenon** [comment]
I am pretty sure this is not a bug with python-shell. "ModuleNotFoundError" is a import error with python, meaning that either the pip package is not installed (if it's a pip package - is it?), python can't find the file, or you are accidently using a different version of python than when you run it on the terminal.

How do you run the script in the terminal? Where is the script located? Where is "telegramcustombot.main" located?

**Comment by LeonardoRick** [comment]
I'm installing over `pip install .` on the root folder of my project containing a `setup.py` file. It works normally wihtout `python-shell` and it works normally too inside `python-shell` when any `env` variable is defined

```
$ pip show telegramcustombot
Name: telegramcustombot
Version: 0.0.1
Summary: Python project to send messages on telegram
Author: Leonardo Rick
Author-email: leonardo@lanytecnologia.com.br
License: Copyright
Location: c:\users\leona\appdata\roaming\python\python38\site-packages
Requires: firebase-admin, python-dateutil, pynacl, cffi, telethon, tgcrypto, pyrogram, apscheduler, python-engineio, python-socketio, rx
Required-by:
```

I'm on `windows` so I have only one python version

```
$ python --version
Python 3.8.7
```

**Comment by Almenon** [comment]
> I'm on windows so I have only one python version

The default python path on windows currently is 'py'. You don't specify `pythonPath` in the options which means it is defaulting to 'py'. I'm guessing `py` is a leftover executable from when you installed a earlier version of python. 

To fix this I suggest specifying "python" as the pythonPath.

**Comment by LeonardoRick** [comment]
Specifying the `pythonPath` throws the same error.

this execute the program:
`const options = { scriptPath: 'C:\\users\\leona\\iq\\iq-websocket\\python\\botR', pythonPath: 'C:\\Program Files\\Python38\\python.exe'};`

this throws the above error:
`const options = { scriptPath: 'C:\\users\\leona\\iq\\iq-websocket\\python\\botR', env: {  PYTHONIOENCODING : 'utf8'}, pythonPath: 'C:\\Program Files\\Python38\\python.exe'};`

**Comment by caleb15** [comment]
What is the output of `which python` on the terminal?

**Comment by Almenon** [comment]
What is the output of which python on the terminal?

**Comment by LeonardoRick** [comment]
```
PS D:\OneDrive\Área de Trabalho> python --version
Python 3.8.7
```
it's on windows

**Comment by Almenon** [comment]
That's the version command - can you run `which python` please?

**Comment by LeonardoRick** [comment]
I guess for windows its `where python`?

```
C:\Users\leona>where python
C:\Program Files\Python38\python.exe
C:\Users\leona\AppData\Local\Microsoft\WindowsApps\python.exe
```

**Comment by Almenon** [comment]
Dang, that does match with the path you specified in the code.... that's really weird :|

HOLY CRAP you're right 0_o

I just reproduced your problem with a slightly different scenario... that's SO BIZZARE. I have no idea how that env var could possibly be .... unless defining `env` overwrites the default env vars ... 🤔 

Yep! That's exactly the case:

https://nodejs.org/api/child_process.html

> env \<Object\> Environment key-value pairs. Default: process.env.


So this is not a bug after all.  It's working as described in the node docs - to add a env var instead of overwriting you need to combine it with process.env.

**Comment by LeonardoRick** [comment]
@Almenon  That's awesome! you discovered the solution while typing, lol.

After reading I found [this stackoverflow](https://stackoverflow.com/a/7825732/10526869) and specific looking to the @DanielSmedegaardBuus comment we can't see he is talking exactly about this and how the documentation do not make clear how important and dangerous is to do that.

So, considering it, I changed the env variable to extend `process.env` as you said:

```python
  env = {
    PYTHONIOENCODING: 'utf8',
    PYTHONUNBUFFERED: 'X',
    ...process.env
  };
```

And now its working perfectly. Case closed.
